platform-dom/src/main/java/com/softicar/platform/dom/elements/input/auto/pattern/MultiPatternMatchRange.java

```diff
@@ -0,0 +1,44 @@
+package com.softicar.platform.dom.elements.input.auto.pattern;
+
+import java.util.Comparator;
+
+public class MultiPatternMatchRange implements Comparable<MultiPatternMatchRange> {
+
+	private final int fromIndex;
+	private final int toIndex;
+
+	MultiPatternMatchRange(int fromIndex, int toIndex) {
+
+		if (fromIndex < 0) {
+			throw new IllegalArgumentException();
+		}
+		if (toIndex < 0) {
+			throw new IllegalArgumentException();
+		}
+		if (toIndex < fromIndex) {
+			throw new IllegalArgumentException();
+		}
```

Comment:
so empty ranges are allowed?

Reply:
They are. An empty range is generated when the matcher is fed an empty pattern.